layouts/partials/section-index.html

```diff
@@ -1,31 +1,33 @@
 <div class="section-index">
-    {{ $parent := .Page }}
-    {{ $pages := (where .Site.Pages "Section" .Section).ByWeight }}
+    {{ $parent := .Page -}}
+    {{ $pages := (where .Site.Pages "Section" .Section).ByWeight -}}
     {{ $pages = (where $pages "Type" "!=" "search") }}
-    {{ $pages = (where $pages ".Params.hide_summary" "!=" true) }}
-    {{ $pages = (where $pages ".Parent" "!=" nil) }}
-    {{ $pages = (where $pages "Parent.File.UniqueID" "==" $parent.File.UniqueID) }}
-    {{ if or $parent.Params.no_list (eq (len $pages) 0) }}
+    {{ $pages = (where $pages ".Params.hide_summary" "!=" true) -}}
+    {{ $pages = (where $pages ".Parent" "!=" nil) -}}
+    {{ with .Parent.File -}}
```

Comment:
Since we're not using the value, I'd opt for `if`:

```suggestion
    {{ if .Parent.File -}}
```

Reply:
Done.